src/lib/structures/ty.ml

```diff
@@ -60,7 +60,7 @@ module Smtlib = struct
     | Trecord { args; name; _ } | Tadt (name, args) ->
       Fmt.(pf ppf "(@[%a %a@])" Uid.pp name (list ~sep:sp pp) args)
     | Tvar { v; value = None; _ } -> Fmt.pf ppf "A%d" v
-    | Tvar { value = Some t; _ } -> pp ppf t
+    | Tvar { value = Some t; _ } -> Fmt.pf ppf "[%a]" pp t
```

Comment:
I think this should be reverted. Types that include variables that are no longer canonical should behave identically to types where the corresponding substitution has been performed: the `value` field here is just an artifact of an efficient implementation of substitutions using mutability; it should not appear in user-facing output (notwithstanding the fact that this syntax is not SMT-LIB compliant).

Reply:
Sure. Actually we should create a printer for debugging only instead of using the SMT-LIB or AE ones depending on the input format. For types, the Dolmen one should be sufficient.